conans/model/options.py

```diff
@@ -95,6 +92,13 @@ def validate(self):
         if None not in self._possible_values:
             raise ConanException("'options.%s' value not defined" % self._name)
 
+    def constrain_possible_values(self, possible_values):
```

Comment:
Can't actually be part of the public api, we don't know at this point if our value is frozen - we need to make this private or freeze-aware